### PR TITLE
CAMEL-12062:  Propagate encoding in charset property from jaxb component

### PR DESCRIPTION
**Pull request by IIlllII**

### CODE REVIEW

File: components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java [hunk 1]
```diff
@@ -136,6 +136,10 @@ public void marshal(Exchange exchange, Object graph, OutputStream stream) throws
             String charset = exchange.getProperty(Exchange.CHARSET_NAME, String.class);
             if (charset == null) {
                 charset = encoding;
+                //Propagate the encoding of the exchange
```
Comment (oscerd) [inline components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java]: This looks a bit ugly.

Comment (oscerd) [reply components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java]: Maybe we can find a better way.

Comment (IIlllII) [reply components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java]: Ok, what is the appropriate way to communicate the encoding?

This is what is done by the file endpoint, and afaik how that is able to accomplish the same feature:
`/**
     * Set up the exchange properties with the options of the file endpoint
     */
    public void configureExchange(Exchange exchange) {
        // Now we just set the charset property here
        if (getCharset() != null) {
            exchange.setProperty(Exchange.CHARSET_NAME, getCharset());
        }
    }`

**Comment by oscerd** [comment]
Thanks the PR has been merged. We'll see if we can do something better in future.

**Pull request closed without merging**